crates/ibc/src/core/context.rs

```diff
@@ -92,34 +90,12 @@ impl std::error::Error for RouterError {
 /// Context to be implemented by the host that provides all "read-only" methods.
 ///
 /// Trait used for the top-level [`validate`](crate::core::validate)
-pub trait ValidationContext: Router {
-    type ClientValidationContext;
-    type E: ClientExecutionContext;
-    type AnyConsensusState: ConsensusState;
-    type AnyClientState: ClientState<Self::ClientValidationContext, Self::E>;
-
-    fn get_client_validation_context(&self) -> &Self::ClientValidationContext;
-    fn get_client_execution_context(&mut self) -> &mut Self::E;
-
-    /// Returns the ClientState for the given identifier `client_id`.
-    ///
-    /// Note: Clients have the responsibility to store client states on client creation and update.
-    fn client_state(&self, client_id: &ClientId) -> Result<Self::AnyClientState, ContextError>;
+pub trait ValidationContext: Router + ClientValidationContext {
```

Comment:
Given my other comments, having `ClientValidationContext` as a supertrait of the core `ValidationContext` no longer works - ultimately batching all the associated types in a `ClientTypes` trait would work if all these traits (core and client's `ValidationContext`/`ExecutionContext`) needed all the types, but they don't